diff --git a/vendor/magento/module-quote/Model/Quote/Address/Total/Shipping.php b/vendor/magento/module-quote/Model/Quote/Address/Total/Shipping.php
index 88edeb5b99d..14930106212 100644
--- a/vendor/magento/module-quote/Model/Quote/Address/Total/Shipping.php
+++ b/vendor/magento/module-quote/Model/Quote/Address/Total/Shipping.php
@@ -56,6 +56,7 @@ class Shipping extends \Magento\Quote\Model\Quote\Address\Total\AbstractTotal

         $address->setWeight(0);
         $address->setFreeMethodWeight(0);
+        $address->setFreeShipping(false);

         $addressWeight = $address->getWeight();
         $freeMethodWeight = $address->getFreeMethodWeight();