diff --git a/vendor/magento/module-catalog-search/Model/Indexer/Fulltext/Action/DataProvider.php b/vendor/magento/module-catalog-search/Model/Indexer/Fulltext/Action/DataProvider.php
index 5a5529c..ea3efd1 100644
--- a/vendor/magento/module-catalog-search/Model/Indexer/Fulltext/Action/DataProvider.php
+++ b/vendor/magento/module-catalog-search/Model/Indexer/Fulltext/Action/DataProvider.php
@@ -95,6 +95,11 @@ class DataProvider
     private $metadata;

     /**
+     * @var \Magento\Framework\App\Config\ScopeConfigInterface
+     */
+    private $config;
+
+    /**
      * @param ResourceConnection $resource
      * @param \Magento\Catalog\Model\Product\Type $catalogProductType
      * @param \Magento\Eav\Model\Config $eavConfig
@@ -123,6 +128,8 @@ class DataProvider
         $this->storeManager = $storeManager;
         $this->engine = $engineProvider->get();
         $this->metadata = $metadataPool->getMetadata(ProductInterface::class);
+        $objectManager = \Magento\Framework\App\ObjectManager::getInstance();
+        $this->config = $objectManager->get(\Magento\Framework\App\Config\ScopeConfigInterface::class);
     }

     /**
@@ -166,6 +173,8 @@ class DataProvider
                 []
             );

+        $this->addOutOfStockFilter($select);
+
         if ($productIds !== null) {
             $select->where('e.entity_id IN (?)', $productIds);
         }
@@ -501,4 +510,37 @@ class DataProvider

         return $value;
     }
+
+    /**
+     * Exclude out of stock products from the list
+     *
+     * @param \Magento\Framework\DB\Select $select
+     */
+    private function addOutOfStockFilter($select)
+    {
+        if (!$this->isOutOfStockVisible()) {
+            $select->distinct(true);
+            $select->joinLeft(
+                ['stock_index' => $this->resource->getTableName('cataloginventory_stock_status')],
+                'e.entity_id = stock_index.product_id',
+                []
+            );
+            $select->where('stock_index.stock_status = ?', \Magento\CatalogInventory\Model\Stock::STOCK_IN_STOCK);
+        }
+    }
+
+    /**
+     * Is out of stock products visible on the frontend
+     *
+     * @return bool
+     */
+    private function isOutOfStockVisible()
+    {
+        $isShowOutOfStock = $this->config->isSetFlag(
+            'cataloginventory/options/show_out_of_stock',
+            \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+        );
+
+        return $isShowOutOfStock === true;
+    }
