diff --git a/vendor/magento/module-catalog/Model/ResourceModel/Product/Collection.php b/vendor/magento/module-catalog/Model/ResourceModel/Product/Collection.php
index 58e8424663c..a2fe2e9323a 100644
--- a/vendor/magento/module-catalog/Model/ResourceModel/Product/Collection.php
+++ b/vendor/magento/module-catalog/Model/ResourceModel/Product/Collection.php
@@ -1657,9 +1657,6 @@ class Collection extends \Magento\Catalog\Model\ResourceModel\Collection\Abstrac
                 $this->getSelect()->order($this->_getAttributeFieldName($attribute) . ' ' . $dir);
                 return $this;
             }
-            if ($this->isEnabledFlat()) {
-                $this->getSelect()->order("cat_index_position {$dir}");
-            }
             // optimize if using cat index
             $filters = $this->_productLimitationFilters;
             if (isset($filters['category_id']) || isset($filters['visibility'])) {
@@ -1968,7 +1965,7 @@ class Collection extends \Magento\Catalog\Model\ResourceModel\Collection\Abstrac
             $this->getSelect()->join(
                 ['cat_index' => $this->getTable('catalog_category_product_index')],
                 $joinCond,
-                ['cat_index_position' => 'position']
+                ['cat_index.position' => 'position']
             );
         }
 
@@ -2008,7 +2005,7 @@ class Collection extends \Magento\Catalog\Model\ResourceModel\Collection\Abstrac
             $this->getSelect()->join(
                 ['cat_pro' => $this->getTable('catalog_category_product')],
                 $joinCond,
-                ['cat_index_position' => 'position']
+                ['cat_pro.position' => 'position']
             );
         }
         $this->_joinFields['position'] = ['table' => 'cat_pro', 'field' => 'position'];
