diff --git a/vendor/magento/module-page-cache/Controller/Block.php b/vendor/magento/module-page-cache/Controller/Block.php
index 3fb095af..c686d638 100644
--- a/vendor/magento/module-page-cache/Controller/Block.php
+++ b/vendor/magento/module-page-cache/Controller/Block.php
@@ -40,7 +40,7 @@ abstract class Block extends \Magento\Framework\App\Action\Action
             return [];
         }
         $blocks = json_decode($blocks);
-        $handles = json_decode($handles);
+        $handles = json_decode(base64_decode($handles));
 
         $this->_view->loadLayout($handles, true, true, false);
         $data = [];
diff --git a/vendor/magento/module-page-cache/Observer/ProcessLayoutRenderElement.php b/vendor/magento/module-page-cache/Observer/ProcessLayoutRenderElement.php
index 4aa54f11..1e3c199a 100644
--- a/vendor/magento/module-page-cache/Observer/ProcessLayoutRenderElement.php
+++ b/vendor/magento/module-page-cache/Observer/ProcessLayoutRenderElement.php
@@ -56,7 +56,7 @@ class ProcessLayoutRenderElement implements ObserverInterface
             'page_cache/block/esi',
             [
                 'blocks' => json_encode([$block->getNameInLayout()]),
-                'handles' => json_encode($layout->getUpdate()->getHandles())
+                'handles' =>  base64_encode(json_encode($layout->getUpdate()->getHandles()))
             ]
         );
         // Varnish does not support ESI over HTTPS must change to HTTP
